lib/src/update.dart

```diff
@@ -201,15 +219,18 @@ linter:
 $output
 ''');
   }
-
-  new File(analysisOptionsFilename).writeAsStringSync(sb.toString());
+  String finalOutput = sb.toString();
+  if (writeToFile) {
+    new File(analysisOptionsFilename).writeAsStringSync(finalOutput);
+  }
```

Comment:
I feel like just returning the value instead of optionally writing it might make the code cleaner. The actual file operation could be totally separate then, and it could even be tested on its own pretty easily.

Reply:
That's true it does feel weird to have a write to file option in a function named writeAnalysisOptions.